docs/admin/cluster-management.md

```diff
@@ -78,20 +101,73 @@ replication controller, then a new copy of the pod will be started on a differen
-    node discovery; currently this is only Google Compute Engine, not including CoreOS on Google Compute Engine using kube-register). See [Node](node.md).
+
+Mark the node to be rebooted as unschedulable:
+
+```console
+kubectl replace nodes $NODENAME --patch='{"apiVersion": "v1", "spec": {"unschedulable": true}}'
+```
+
+This keeps new pods from landing on the node while you are trying to get them off.
+
+Get the pods off the machine, via any of the following strategies:
+   * Wait for finite-duration pods to complete.
+   * Delete pods with:
+
+```console
+kubectl delete pods $PODNAME
+```
+
+For pods with a replication controller, the pod will eventually be replaced by a new pod which will be scheduled to a new node. additionally, if the pod is part of a service, then clients will automatically be redirected to the new pod.
+
+For pods with no replication controller, you need to bring up a new copy of the pod, and assuming it is not part of a service, redirect clients to it.
+
+Perform maintainence work on the node.
+
+Make the node schedulable again:
+
+```console
+kubectl replace nodes $NODENAME --patch='{"apiVersion": "v1", "spec": {"unschedulable": false}}'
+```
+
+If you deleted the node's VM instance and created a new one, then a new schedulable node resource will
+be created automatically when you create a new VM instance (if you're using a cloud provider that supports
+node discovery; currently this is only Google Compute Engine, not including CoreOS on Google Compute Engine using kube-register). See [Node](node.md) for more details.
+
+## Advanced Topics
+
+### Upgrading to a different API version
+
+When a new API version is released, you may need to upgrade a cluster to support the new API version (e.g. switching from 'v1' to 'v2' when 'v2' is launched)
+
+This is an infrequent event, but it requires careful management. There is a sequence of steps to upgrade to a new API version.
+
+   1. Turn on the new api version.
+   1. Upgrade the cluster's storage to use the new version.
+   1. Upgrade all config files. Identify users of the old API version endpoints.
+   1. Update existing objects in the storage to new version by running `cluster/update-storage-objects.sh`.
+   1. Turn off the old version.
+
+### Turn on or off an API version for your cluster
+
+Specific API versions can be turned on or off by passing --runtime-config=api/<version> flag while bringing up the API server. For example: to turn off v1 API, pass `--runtime-config=api/v1=false`.
+runtime-config also supports 2 special keys: api/all and api/legacy to control all and legacy APIs respectively.
+For example, for turning off all api versions except v1, pass `--runtime-config=api/all=false,api/v1=true`.
+For the purposes of these flags, _legacy_ APIs are those APIs which have been explicitly deprecated (e.g. `v1beta3`).
+
+### Switching your cluster's storage API version
```

Comment:
You might want to explain what this does. (It's a little confusing that "switching your cluster's storage API version" is not the same as "update existing objects in the storage to new version").